infra/modules/iam/main.tf

```diff
@@ -20,112 +20,236 @@ resource "aws_iam_group_policy" "wic-prp-eng" {
+      "ecr:ListTagsForResource",
+    ]
+    resources = [
+      "arn:aws:ecr:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:repository/*",
+    ]
   }
   statement {
-    sid    = "IAMServices"
+    sid    = "ELB"
+    effect = "Allow"
+    actions = [
+      "elasticloadbalancing:AddTags",
+      "elasticloadbalancing:CreateListener",
+      "elasticloadbalancing:CreateLoadBalancer",
+      "elasticloadbalancing:CreateRule",
+      "elasticloadbalancing:CreateTargetGroup",
+      "elasticloadbalancing:DeleteRule",
+      "elasticloadbalancing:DeleteListener",
+      "elasticloadbalancing:DeleteLoadBalancer",
+      "elasticloadbalancing:DeleteTargetGroup",
+      "elasticloadbalancing:ModifyLoadBalancerAttributes",
+      "elasticloadbalancing:ModifyTargetGroupAttributes",
+      "elasticloadbalancing:SetSecurityGroups",
+    ]
+    resources = [
+      "arn:aws:elasticloadbalancing:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:targetgroup/*",
+      "arn:aws:elasticloadbalancing:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:loadbalancer/*",
+      "arn:aws:elasticloadbalancing:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:listener/*",
+      "arn:aws:elasticloadbalancing:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:listener-rule/*",
+    ]
+  }
+  statement {
+    sid    = "IAM"
     effect = "Allow"
     actions = [
       "iam:AttachRolePolicy",
       "iam:CreateRole",
-      "iam:CreateServiceLinkedRole",
-      "iam:GenerateServiceLastAccessedDetails",
+      "iam:DeleteRole",
+      "iam:DeleteRolePolicy",
+      "iam:DetachRolePolicy",
       "iam:GetRole",
       "iam:GetRolePolicy",
-      "iam:GetServiceLinkedRoleDeletionStatus",
       "iam:ListAttachedRolePolicies",
+      "iam:ListInstanceProfilesForRole",
       "iam:ListRolePolicies",
-      "iam:PutRolePolicy"
+      "iam:PassRole",
+      "iam:PutRolePolicy",
+      "iam:TagRole",
+    ]
+    resources = [
+      "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/*"
     ]
-    resources = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/*"]
   }
   statement {
-    sid       = "KMSCreate"
-    effect    = "Allow"
-    actions   = ["kms:CreateAlias"]
-    resources = ["arn:aws:kms:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:alias/*"]
+    sid    = "IAMPolicy"
+    effect = "Allow"
+    actions = [
+      "iam:CreatePolicy",
+      "iam:DeletePolicy",
+      "iam:GetPolicy",
+      "iam:GetPolicyVersion",
+      "iam:ListPolicyVersions",
+      "iam:TagPolicy",
+    ]
+    resources = [
+      "arn:aws:iam::${data.aws_caller_identity.current.account_id}:policy/*"
+    ]
   }
   statement {
-    sid    = "KMSServices"
+    sid    = "KMS"
     effect = "Allow"
     actions = [
+      "kms:DescribeKey",
       "kms:CreateAlias",
       "kms:Decrypt",
```

Comment:
This list of actions doesn't allow encryption.. 
Are y'all sure this set of perms will allow someone to put new files into the S3 bucket, with encryption using that KMS key?

Reply:
Good question! I'm not sure. I know these permissions allow a user (one of us) in the user group to create and destroy environments, including the S3 buckets. 

Do we need to put files into the S3 bucket? Is this in reference to how we're going to handle doc upload?